onyo/lib/commands.py

```diff
@@ -963,29 +971,39 @@ def onyo_unset(inventory: Inventory,
     ----------
     inventory: Inventory
         The Inventory in which to unset key/values for assets.
-    keys: list
+    keys: list of str
         The keys that will be unset in assets.
         If keys do not exist in an asset, a debug message is logged.
         If keys are specified which appear in asset names an error is raised.
+        If `keys` is empty an error is raised.
     match: list of Callable, optional
       Callables suited for use with builtin `filter`. They are
       passed an asset dictionary and expected to return a `bool`,
       where `True` indicates a match. `keys` will be removed from
       all assets that are matched by all callables in this list.
-    paths: Path or list of Path, optional
+    paths: list of Path, optional
         Paths to assets or directories for which to unset key-value pairs.
         If paths are directories, the values will be unset recursively in assets
         under the specified path.
-        If no paths are specified, CWD is used as default.
+        If no paths are specified, the inventory root is used as default.
     depth: int
         Depth limit of recursion if a `path` is a directory.
         0 means no limit and is the default.
     message: str, optional
         An optional string to overwrite Onyo's default commit message.
-    """
-    paths = paths or []
 
-    non_inventory_paths = [str(p) for p in paths
+    Raises
+    ------
+    ValueError
+        If paths are invalid, or `keys` are empty or invalid.
+
+    """
+    paths = paths or [inventory.root]
+    if not keys:
+        raise ValueError("At least one key must be specified.")
+    if not all(keys):
+        raise ValueError("Keys are not allowed to be empty or None-values.")
```

Comment:
I think this should not be necessary in case of unset. If there's no empty key in a matching asset, then we should just end up failing because we are trying to unset a key that's not there in the first place, not because the key is empty. If such a key exists for whatever reason, then we should be able to unset it (if only to fix the situation).